modules/renter/contractor/contractor.go

```diff
@@ -17,6 +17,11 @@ var (
 	errNilTpool  = errors.New("cannot create contractor with nil transaction pool")
 )
 
+type cachedRevision struct {
+	revision    types.FileContractRevision
+	merkleRoots []crypto.Hash
```

Comment:
100MB of Merkle roots would mean a 13 TB file. I don't think we need to worry about that for a while.
It is true, though, that caching a revision for every file effectively doubles the data that the renter stores locally. So a diff is reasonable. I'll leave a TODO.